Library/Homebrew/dev-cmd/tests.rb

```diff
@@ -87,6 +87,7 @@ def tests
         ENV.delete(env)
       end
 
+      ENV["HOMEBREW_TEST"] = "1"
```

Comment:
Is this a reasonable way to fix the tests? I set `HOMEBREW_TEST` only when `brew tests` is being run. Then, intentionally don't output the warning about turning developer mode on if `HOMEBREW_TEST` is set.

Reply:
@Rylan12 I think not unfortunately. Either add a documented environment variable to bypass this, fix the tests to expect this output or only test the case that doesn't output this message.

Reply:
I added a `HOMEBREW_NO_DEV_CMD_MESSAGE` environment variable to be set automatically by `brew tests`. How's that, instead?